### PR TITLE
Adding api calls

### PR DESCRIPTION
**Pull request by akash-1318**
# Added API calls
I have added api calls for login , signup , wishlist and cart page

# files to check 
1. src/contexts/auth-context.js
2. src/pages/login-page/Login.js
3. src/pages/signup-page/SignUp.js
4. src/pages/cart/cart-functions.js
5. src/pages/wishlist/wishlist-functions.js

https://user-images.githubusercontent.com/64393554/161815679-b3a5f083-b88f-4035-a34a-a7bbadf01f92.mov

### CODE REVIEW

File: src/pages/signup-page/SignUp.js [hunk 1]
```diff
@@ -1,31 +1,91 @@
...
         </div>
         <div className="cred__input-field password__field">
           <label>Password</label>
-          <input placeholder="password" type="password"></input>
+          <input 
+          placeholder="password" 
+          type="password"
+          onChange={(e) => setSignupCred({...signupCred, password : e.target.value})}
+          ></input>
           <i className="bx bx-show"></i>
```
Comment (irajbhinde) [inline src/pages/signup-page/SignUp.js:80]: When time permits you can implement the show password functionality here, won't take much time

File: src/reducers/cart-reducer.js [hunk 2]
```diff
@@ -2,13 +2,17 @@ const cartReducer = (state, action) => {
...
         case "ADD_TO_CART":
-            return {...state, cartProducts : [...state.cartProducts, {...payload, quantity : 1}]}
+            return {...state, cartProducts : payload}
         case "REMOVE_FROM_CART":
             return {...state, cartProducts : [...state.cartProducts.filter((cartItem) => cartItem._id !== payload._id)]}
-        case "INCREMENT":
-            return {...state, cartProducts : state.cartProducts.map((qtItem) => qtItem._id === payload._id ? {...qtItem, quantity : qtItem.quantity + 1} : qtItem)}
-        case "DECREMENT":
-            return {...state, cartProducts : state.cartProducts.map((qtItem) => qtItem._id === payload._id ? {...qtItem, quantity : qtItem.quantity > 1 ? qtItem.quantity - 1 : qtItem.quantity} : qtItem)}
+        // case "INCREMENT":
```
Comment (irajbhinde) [inline src/reducers/cart-reducer.js:8]: You can remove commented code here

File: src/components/navigation/navigation.js [hunk 3]
```diff
@@ -1,16 +1,33 @@
...
   const {dispatch} = useProductContext()
   const {wishState} = useWishContext()
   const {cartState} = useCartContext()
   const {cartProducts} = cartState
-  const qunatityReducer = (previousProd, currentProd) => currentProd.quantity + previousProd
+  const qunatityReducer = (previousProd, currentProd) => currentProd.qty + previousProd
   const cartProductQuantity = cartProducts.reduce(qunatityReducer, 0)
+
+  const navNavigation = (type) => {
+    type === "wishlist" ? (authCred.authStatus ? (navigate('/wishlist')):(navigate('/login'))) : (authCred.authStatus ? (navigate('/cart')):(navigate('/login')))
```
Comment (irajbhinde) [inline src/components/navigation/navigation.js:19]: follow a consistent naming convention for type, here it is small letters & at few places you have used CAPS

**Review by irajbhinde — COMMENTED** [review]
Apart from single line comments - For better UX, you can add a remove from wishlist button on your wishlist page, in deploy preview it was confusing for me to remove the product from wishlist.

Apart from that code looks good, keep up the good work

**Comment by akash-1318** [comment]
> Apart from single line comments - For better UX, you can add a remove from wishlist button on your wishlist page, in deploy preview it was confusing for me to remove the product from wishlist.
> 
> Apart from that code looks good, keep up the good work

Will surely work on your suggestions thanks for the review

File: src/components/product-card/ProductCard.js [hunk 4]
```diff
@@ -44,15 +58,26 @@ export default function ProductCard({ product }) {
...
+              <button className="btn solid__primary prod__card-btn" 
+              // onClick={() => cartDispatch({type : "REMOVE_FROM_CART", payload : product})}
+              onClick = {() => removeFromCart(product, cartDispatch, authToken)}
+              >
               Remove From Cart
             </button>
             ) : (
-              <button className="btn solid__secondry prod__card-btn" onClick={() => cartDispatch({type : "ADD_TO_CART", payload : product})}>
+              <button className="btn solid__secondry prod__card-btn" 
+              // onClick={() => cartDispatch({type : "ADD_TO_CART", payload : product})}
```
Comment (darshandamre) [inline src/components/product-card/ProductCard.js:69]: remove comments if not needed

File: src/contexts/auth-context.js [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import {createContext, useContext, useState,} from 'react'
+
+const AuthContext = createContext()
+
+const AuthContextProvider = ({children}) => {
+    const [authCred, setAuthCred] = useState({
+        authToken : localStorage.getItem("TOKEN"),
+        authStatus : localStorage.getItem("TOKEN") ? true : false,
```
Comment (darshandamre) [inline src/contexts/auth-context.js:8]: here you can use NOT operator twice, like this
```js
authStatus : !!localStorage.getItem("TOKEN")
```
this converts a value to boolean type

File: src/pages/cart/cart-functions.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    } catch(err){
+      console.log(err)
+    }
+  }
+
+  const updateQty = async (type, product, cartDispatch, authToken) => {
+      try{
+          const incrementData = await axios.post(`/api/user/cart/${product._id}`,
+          {
+              action : { type: type }
```
Comment (darshandamre) [inline src/pages/cart/cart-functions.js:35]: here you can directly write
```js
{
  action: { type }
}
```

**Pull request merged**